### PR TITLE
Lauren Cardella -- Carets

### PR DESCRIPTION
**Pull request by enigmagnetic**
# API Muncher
Congratulations! You're submitting your assignment!

## Comprehension Questions
| Question | Answer    |
| :------------- | :------------- |
| How did you go about exploring the Edamam API, how did you try querying the API? | I read the documentation, then tried two different queries in Postman, one using the q key (for a recipe search by keyword) and one using the r key (to return a recipe by ID). |
| Describe your API Wrapper.  How did you decide on the methods you created? | I created two methods for my API Wrapper, one to return a list of search results (used in the recipes controller index action) and the other to return a particular recipe (used in the recipes controller show action). |
| Describe an edge case or failure case test you wrote for your API Wrapper. | I wrote an edge case test for both API Wrapper methods to make sure that an invalid argument returns an empty array. |
| Explain how VCR aids in testing an API. | VCR records the API responses for a given call so that it can be used in subsequent matching calls. This helps reduce the number of calls made to the API (so that the tests don't contribute to any limits), speeds the tests up because new calls are not needed for each test, and it allows for testing even when the API service is down. |
| What is the Heroku URL of your deployed application? | https://muncher-ada.herokuapp.com/ |

### CODE REVIEW

File: lib/edamam_api_wrapper.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    end
+  end
+
+  def self.find_recipe(uri, token = nil)
+    token ||= TOKEN
+    url = BASE_URL + "search?r=#{uri}&app_id=#{APP_ID}&app_key=#{token}"
+
+    data = HTTParty.get(url)
+
+    unless data.empty?
```
Comment (CheezItMan) [inline lib/edamam_api_wrapper.rb:30]: I would add a rescue because for bad requests data is coming back as `[` which won't parse correctly.

File: app/views/recipes/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<h2 class="recipe-index"><%= params[:query].capitalize %> Recipes</h2>
+
+<div class="recipe-index">
+<% if @recipes.empty? %>
+  <h4>There are no recipes matching your search.</h4>
+<% else %>
+  <% @recipes.each do |recipe| %>
+    <article class="recipe">
+      <%= image_tag (recipe.image) %>
```
Comment (CheezItMan) [inline app/views/recipes/index.html.erb:9]: I would make the images for the results links as it's natural to want to click on'em.

**Comment by CheezItMan** [comment]
# API Muncher
## What We're Looking For

| Feature | Feedback    |
| :------------- | :------------- |
| **Core Requirements** |   |
| Git hygiene | Good number of commits and good commit messages |
| Comprehension questions	| Check |
| **General** |  |
| Rails fundamentals (RESTful routing, use of named paths) | Check |
| Semantic HTML | Check, although your index page has a `div` that could be something else |
| Errors are reported to the user | No search results reported for invalid searches.  It crashes for invalid URIs for individual recipes |
| API Wrapper to handle the API requests | Check, although you need to work on error handling |
| Controller testing | 2 failing tests |
| Lib testing | Check, but a failing test due to lack of error handling in your wrapper |
| Search Functionality | Check |
| List Functionality | Check |
| Show individual item functionality (link to original recipe opens in new tab) | Check |
| **Styling** |  |
| Foundation Styling for responsive layout | Well done |
| List View shows 10 items at a time/pagination | Check |
| The app is styled to create an attractive user interface | Check |
| **API Features** | |
| The App attributes Edaman | Yes, but no link.  |
| The VCR casettes do not contain the API key | Check |
| **External Resources** |  |
| Link to deployed app on Heroku | Doesn't work!  |
|  **Overall** | Your Heroku isn't working and you have trouble with negative case tests as well as some error handling in your wrapper.  Not bad, but could be better. |

File: test/controllers/recipes_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+require "test_helper"
+
+class RecipesControllerTest < ActionDispatch::IntegrationTest
+
+  describe "index" do
+    it "can get a list of recipe search results" do
+      VCR.use_cassette("index_action") do
+        get recipes_path
+        must_respond_with :success
```
Comment (CheezItMan) [inline test/controllers/recipes_controller_test.rb:9]: If there's not a search here it redirects back to home.

File: test/controllers/recipes_controller_test.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      end
+    end
+  end
+
+  describe "show" do
+    it "can get a specific recipe" do
+      VCR.use_cassette("show_action") do
+        get recipe_path
```
Comment (CheezItMan) [inline test/controllers/recipes_controller_test.rb:17]: There's no recipe to show, so it redirects to the homepage, at least I'd hope so, or prints an error message.
